src/libs/http/http-deepl-client/src/lib/deepl-client.ts

```diff
@@ -315,42 +181,70 @@ export class DeeplClient {
     return targetLanguages;
   }
 
-  /**
-   * Retrieves the preserve formatting setting from the configuration.
-   * @returns {boolean} The preserve formatting setting.
-   */
+  private getTargetLanguage(requestedTargetLanguage: string): string {
+    const shouldUseSimplifiedTargetLanguage = ![
+      'en-US',
+      'en-GB',
+      'pt-BR',
+      'pt-PT',
+      'zh-Hans',
+    ].includes(requestedTargetLanguage);
+
+    if (shouldUseSimplifiedTargetLanguage) {
+      const languageRegex = new RegExp('([a-z]{2})(?:-[A-Z]{2})?');
+      const simplifiedTargetLanguage =
+        requestedTargetLanguage.match(languageRegex);
+
+      if (!simplifiedTargetLanguage) {
+        throw new Error(
+          `Invalid target language code. ${requestedTargetLanguage}`
+        );
+      }
+
+      return simplifiedTargetLanguage[1];
+    }
+
+    return requestedTargetLanguage === 'en' ? 'en-GB' : requestedTargetLanguage;
+  }
+
+  private isSupportedTargetLanguage(targetLanguage: string): boolean {
+    return DeeplClient.supportedTargetLanguages
+      .map(x => x.code)
+      .includes(targetLanguage as deepl.LanguageCode);
+  }
+
+  private getFormality(targetLanguage: string): deepl.Formality | undefined {
```

Comment:
Language related stuff should be set during initialization. And when a new language is added.